File: cpp/1572_Matrix_Diagonal_Sum.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int n = mat.size(), ans = 0;
        for (int c = 0; c < n; c++) {
            ans += mat[c][c];
            int r2 = n - c - 1;
            if (r2 != c) {
                ans += mat[r2][c];
            }
        }
        return ans;
    }
};

File: python/1572_Matrix_Diagonal_Sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        ans = 0
        n = len(mat)
        for c in range(n):
            ans += mat[c][c]
            r2 = n - c - 1
            if r2 != c:
                ans += mat[r2][c]
        return ans